### ISSUE TITLE
Risky tests in master

### ISSUE THREAD

**Issue by Majkl578** [schmittjoh/serializer#910]
There are currently 9 risky tests that don't assert anything:
```
There were 9 risky tests:

1) JMS\Serializer\Tests\Handler\ArrayCollectionHandlerTest::testSerializeArray
This test did not perform any assertions

2) JMS\Serializer\Tests\Handler\ArrayCollectionHandlerTest::testSerializeArraySkipByExclusionStrategy
This test did not perform any assertions

3) JMS\Serializer\Tests\Handler\DateHandlerTest::testSerializeDate with data set #0 (array('Y-m-d'))
This test did not perform any assertions

4) JMS\Serializer\Tests\Handler\DateHandlerTest::testSerializeDate with data set #1 (array('Y-m-d', '', 'Y-m-d|'))
This test did not perform any assertions

5) JMS\Serializer\Tests\Handler\DateHandlerTest::testSerializeDate with data set #2 (array('Y-m-d', '', 'Y'))
This test did not perform any assertions

6) JMS\Serializer\Tests\Serializer\GraphNavigatorTest::testNavigatorChangeTypeOnSerialization
This test did not perform any assertions

7) JMS\Serializer\Tests\Serializer\JsonSerializationTest::testFormErrorsWithNonFormComponents
This test did not perform any assertions

8) JMS\Serializer\Tests\Serializer\XmlSerializationTest::testWhitelistedDocumentTypesAreAllowed
This test did not perform any assertions

9) JMS\Serializer\Tests\Serializer\XmlSerializationTest::testFormErrorsWithNonFormComponents
This test did not perform any assertions
```

**Comment by goetas** [comment]
True... test should be improved

**Comment by goetas** [comment]
most of the risky tests are because they are testing that exceptions are not thrown... ideas here?

**Comment by Majkl578** [comment]
https://github.com/sebastianbergmann/phpunit/issues/2484

You can use `@doesNotPerformAssertions`, but it would be better to rework the test to actually test some behavior.

**Comment by goetas** [comment]
Do not know what is better... `@doesNotPerformAssertions` or https://github.com/schmittjoh/serializer/pull/918 :thinking:

**Comment by Majkl578** [comment]
I'd say `@doesNotPerformAssertions`. Asserting that true is true is nonsense. :D

**Comment by goetas** [comment]
fixed in #918

**Comment by simPod** [comment]
It doesn't count towards coverage tho
